field/koalabear/vortex/params.go

```diff
@@ -2,12 +2,63 @@ package vortex
 
 import (
 	"errors"
+	"hash"
 
 	"github.com/consensys/gnark-crypto/field/koalabear"
 	"github.com/consensys/gnark-crypto/field/koalabear/fft"
 	"github.com/consensys/gnark-crypto/field/koalabear/sis"
 )
 
+var (
+	ErrWrongSizeHash = errors.New("the hash size should be 32 bytes")
+)
+
+// NewHash a functions returning a hash. Hash functions are stored this way, to allocate
+// them when needed and parallelise the execution when possible.
+type NewHash = func() hash.Hash
+
+// Configuration options of the vortex prover
+type Config struct {
+	// hash function used to build the Merkle tree. By default, this hash is poseidon2.
+	merkleHashFunc NewHash
+	// hash function used to hash the stacked codewords. By default, this hash function is SIS.
+	otherThanSis NewHash
+}
+
+// Option provides options for altering the default behavior of the vortex prover.
+// See the descriptions of the functions returning instances of this
+// type for available options.
+type Option func(opt *Config) error
+
+// WithMerkleHash specifies the hash function used to build the Merkle tree of the hashed
+// columns of the stacked codewords.
+func WithMerkleHash(h hash.Hash) Option {
+	return func(opt *Config) error {
+		bs := h.Size()
+		if bs != 32 {
+			return ErrWrongSizeHash
+		}
+		opt.merkleHashFunc = func() hash.Hash { return h }
+		return nil
+	}
+}
+
+// WithNoSis specifies the hash function used to hash the columns of the stacked codewords.
+func WithNoSis(h hash.Hash) Option {
```

Comment:
maybe instead of `WithNoSis` and `opt.otherThanSis` call them `WithColumnHash` and `opt.columnHash` ?